### PR TITLE
feat: modularize checksums

### PR DESCRIPTION
**Pull request by dayaffe**
New modularization package:
- AWSSDKChecksums

## Issue \#
#1532 

## Description of changes
- AWSSDKChecksums contains details specific to the AWS part of checksums. As of now this includes the method to add the aws-chunked header. AWSSDKChecksums can only contain checksum and chunking related methods that are specific to AWS and are not dependencies of other parts of the smithy-swift package.
- Adds additional dependencies from smithy-swift including SmithyChecksums and SmithyStreams
- Protocol tests now depend on SmithyStreams and a symbol has been added for SmithyStreams.BufferedStream.
- FlexibleChecksums Request/Response middlewares have been moved from smithy-swift to aws-sdk-swift

## Conventional Commits

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Sources/Core/AWSSDKHTTPAuth/AWSSigV4Signer.swift [hunk 1]
```diff
@@ -255,3 +256,14 @@ public class AWSSigV4Signer: SmithyHTTPAuthAPI.Signer {
         }
     }
 }
+
+extension SigningConfig {
```
Comment (dayaffe) [inline Sources/Core/AWSSDKHTTPAuth/AWSSigV4Signer.swift:260]: this method doesnt rely on checksums

File: Sources/Core/AWSSDKHTTPAuth/AWSSigV4Signer.swift [hunk 2]
```diff
@@ -30,6 +30,7 @@ import struct SmithyHTTPAuthAPI.SigningFlags
 import struct Foundation.Date
 import struct Foundation.TimeInterval
 import struct Foundation.URL
+import AWSSDKChecksums
```
Comment (dayaffe) [inline Sources/Core/AWSSDKHTTPAuth/AWSSigV4Signer.swift:33]: the signer still needs to depend on AWSSDKChecksums for setChunkedBody method

File: AWSSDKSwiftCLI/Sources/AWSSDKSwiftCLI/Resources/Package.Base.swift [hunk 3]
```diff
@@ -22,6 +22,7 @@ extension Target.Dependency {
     static var awsSDKEventStreamsAuth: Self { "AWSSDKEventStreamsAuth" }
     static var awsSDKHTTPAuth: Self { "AWSSDKHTTPAuth" }
     static var awsSDKIdentity: Self { "AWSSDKIdentity" }
+    static var awsSDKChecksums: Self { "AWSSDKChecksums" }
```
Comment (sichanyoo) [inline AWSSDKSwiftCLI/Sources/AWSSDKSwiftCLI/Resources/Package.Base.swift:25]: Question: Is a separate module actually necessary for just a single extension with two methods?

**Review by sichanyoo — COMMENTED** [review]
Question on need for AWSSDKChecksums module

File: AWSSDKSwiftCLI/Sources/AWSSDKSwiftCLI/Resources/Package.Base.swift (hunk 3 above)
Comment (dayaffe) [reply AWSSDKSwiftCLI/Sources/AWSSDKSwiftCLI/Resources/Package.Base.swift:25]: I wish more could be moved to AWSSDKChecksums but that's dependent on CRT HTTP Client being refactored in the future. I'd say even if its only two methods its good to have but I don't feel strongly. `aws-chunked` is pretty explicitly aws-only though and one of those methods adds it to the headers

Comment (sichanyoo) [reply AWSSDKSwiftCLI/Sources/AWSSDKSwiftCLI/Resources/Package.Base.swift:25]: I see. We could keep it the way it is in this PR, or move it into AWSClientRuntime. For both cases we would revisit this down the line. I don't feel strongly about either option. Unless Josh has preference for one over the other I suppose we can keep `AWSSDKChecksums`.

**Review by jbelkins — APPROVED** [review]

**Pull request merged**